src/shell_op/shell_op.c

```diff
@@ -6,17 +6,18 @@
 /*   By: yoav <yoav@student.42.fr>                  +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/09/13 16:52:42 by yoav              #+#    #+#             */
-/*   Updated: 2022/10/06 09:41:53 by yoav             ###   ########.fr       */
+/*   Updated: 2022/10/12 11:45:18 by yoav             ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "shell_op.h"
 
-t_error_code	shell_op_create(t_shell_op **ret)
+t_error_code	shell_op_create(t_shell_op **ret, char **envp)
 {
 	*ret = ft_calloc(1, sizeof(t_shell_op));
 	if (!(*ret))
 		return (ALLOCATION_ERROR);
+	(*ret)->envp = envp;
```

Comment:
It is better to create a copy of envp so we can free and manipulate it however we want. I created an utillity for that at ENV, so can can call it directly from here (ill modify it locally once i merged and upload a PR with ENV and this change)